CMakePresets.json

```diff
@@ -0,0 +1,80 @@
+{
+    "version"   : 3,
+    "configurePresets" : [
+        {
+            "name"          : "default",
+            "binaryDir"     : "${sourceDir}/build-with-${presetName}",
+            "cacheVariables" : {
+                "CMAKE_EXPORT_COMPILE_COMMANDS"     : "ON",
+                "CMAKE_BUILD_TYPE"                  : "Release",
+                "CMAKE_CXX_STANDARD"                : "17",
+                "KokkosTools_REUSE_KOKKOS_COMPILER" : "ON",
```

Comment:
If you are going to use that "reuse cxx compiler option", what is the point of setting the cxx compiler below?

Reply:
As discussed, we are not even sure that `KokkosTools_REUSE_KOKKOS_COMPILER` works (as it does not set variables with `FORCE` for instance), and as we already set the compiler above it shouldn't be necessary.